nameserver/dns.go

```diff
@@ -92,3 +96,18 @@ func getMaxReplyLen(r *dns.Msg, proto dnsProtocol) int {
 	}
 	return maxLen
 }
+
+// shuffleAnswers reorders answers for very basic load balancing
+func shuffleAnswers(answers []dns.RR) []dns.RR {
+	n := len(answers)
+	if n > 1 {
+		rand.Seed(time.Now().UTC().UnixNano())
```

Comment:
This is typically done only once per program instantiation. You might want to think about abstracting it too (as you have done for the clock) so that you can have deterministic behaviour in your unit tests...